substrate/client/network/src/litep2p/mod.rs

```diff
@@ -259,6 +259,17 @@ impl Litep2pNetworkBackend {
 		Ok((local_identity, local_peer_id))
 	}
 
+	/// Fetch the number of connected peers from the peerset handle and update
+	/// the atomic `num_connected` shared between the network backend.
+	fn fetch_connected_peers(&self) -> usize {
+		let num_connected_peers = self
+			.peerset_handles
+			.get(&self.block_announce_protocol)
```

Comment:
I don't get why we use the number of connected peers from a specific protocol? I would assume that `libp2p` is not doing this and instead takes the number of all the connected peers?